react_app/src/components/App.js

```diff
@@ -19,11 +21,29 @@ class App extends Component {
 
       newUser: false,
 
+      /**
+       * don't put functions in your state! I am guessing that
+       * the intention of this is that you want to store the whole
+       * app state here in App.js, pass that into Context AS WELL AS
+       * passing a way to update it - so if you want to do that then you
+       * don't need a function here, you can just do this;
+       *
+       *   <AppContext.Provider value={update: this.setState, ...this.state}>
+       *
+       * This approach isn't going to scale, but maybe you don't need it to (yet).
+       * It's also really inefficient (your whole application is going to re-render
+       * every time any piece of the state updates) - again, maybe that's not a
+       * concern yet.
+       */
```

Comment:
... to my shame I just can't get to grips with declaring functions inside calls. I get that its a JS thing, but it scares the pants off me